iceoryx_utils/include/iceoryx_utils/internal/units/duration.inl

```diff
@@ -71,211 +126,298 @@ inline constexpr Duration::Duration(const struct itimerspec& value)
+    double secondsFraction = modf(result, &resultSeconds);
+    return Duration(static_cast<uint64_t>(resultSeconds), 0U)
+           + Duration::nanoseconds(static_cast<uint64_t>(secondsFraction * 1000000000U));
 }
 
-inline namespace duration_literals
-{
-inline constexpr Duration operator"" _ns(long double value)
+template <typename T>
+inline constexpr Duration
+Duration::multiplyNanoseconds(const uint32_t nanoseconds,
+                              const std::enable_if_t<!std::is_floating_point<T>::value, T>& right) const
 {
-    return Duration{value / 1000000000.0};
+    // specialization is needed to prevent a clang warning if `right` is a signed integer and not casted to unsigned
+    // operator*(...) takes care of negative values for right
+    return Duration::nanoseconds(nanoseconds * static_cast<uint64_t>(right));
 }
 
-inline constexpr Duration operator"" _ns(unsigned long long int value) // PRQA S 48
+template <typename T>
+inline constexpr Duration
+Duration::multiplyNanoseconds(const uint32_t nanoseconds,
+                              const std::enable_if_t<std::is_floating_point<T>::value, T>& right) const
 {
-    return Duration{static_cast<long double>(value) / 1000000000.0};
+    // operator*(...) takes care of negative values for right
+    return Duration::nanoseconds(static_cast<uint64_t>(nanoseconds * right));
 }
 
-inline constexpr Duration operator"" _us(long double value)
+template <typename T>
+inline constexpr Duration Duration::operator*(const T& right) const
 {
-    return Duration{value / 1000000.0};
-}
+    static_assert(std::is_arithmetic<T>::value, "non arithmetic types are not supported for multiplication");
 
-inline constexpr Duration operator"" _us(unsigned long long int value) // PRQA S 48
-{
-    return Duration{static_cast<long double>(value) / 1000000.0};
-}
+    if (right < static_cast<T>(0))
+    {
+        return Duration{0U, 0U};
+    }
 
-inline constexpr Duration operator"" _ms(long double value)
-{
-    return Duration{value / 1000.0};
-}
+    /// @todo decide if we want an overflow or saturation
 
-inline constexpr Duration operator"" _ms(unsigned long long int value) // PRQA S 48
-{
-    return Duration{static_cast<long double>(value) / 1000.0};
+    return multiplySeconds<T>(m_seconds, right) + multiplyNanoseconds<T>(m_nanoseconds, right);
 }
 
-inline constexpr Duration operator"" _s(long double value)
+template <typename T>
+inline constexpr Duration Duration::operator/(const T& right) const
 {
-    return Duration{value};
+    static_assert(std::is_arithmetic<T>::value, "non arithmetic types are not supported for multiplication");
+
+    if (right < static_cast<T>(0))
+    {
+        return Duration{0U, 0U};
+    }
+
+    auto result = m_seconds / right;
+    double seconds{0.0};
+    double secondsFraction = modf(result, &seconds);
+    auto nanoseconds =
+        static_cast<uint64_t>(secondsFraction * 1000000000U) + static_cast<uint64_t>(m_nanoseconds / right);
```

Comment:
I am disappointed, so disappointed!!! ;)

What happens when `right == 0`??

Reply:
Which division by zero? There is no `operator/` for `Duration` 😉